### ISSUE TITLE
Somehow let Reek examine ruby files that don't have the .rb extension

### ISSUE THREAD

**Issue by mvz** [troessner/reek#721]
Reek currently won't examine your Rakefile or executables (bin/whatever) because they don't have the `.rb` extension. We should do something smart here.

**Comment by troessner** [comment]
```Bash
file Rakefile

Rakefile: ASCII text
```

No way of doing that without shebang. 
I see 3 possibilities here:

1.) We expect ruby files to have a shebang and then use `file` or something similar
2.) We use something like this: https://github.com/minad/mimemagic (which would add a dependency to `reek`)
3.) We have a whitelist for common files like "Rakefile" and so on

**Comment by mvz** [comment]
I would say we use at least option 3, and then either 1 or 2. It looks like mimemagic implements at least the shebang method.

I'd also have a look at how Rubocop does this.

**Comment by chastell** [comment]
:+1: for whitelisting `Rakefile` and `Gemfile` – the latter until everybody finally switches to `gems.rb` ;) – and I’m ok with using mimemagic as well (I’d prefer this over shelling out to `file`).

**Comment by mvz** [comment]
> the latter until everybody finally switches to gems.rb

This will be a while since even generating a new gem project with bundler will create a Gemfile rather than gems.rb :crying_cat_face:.

**Comment by chastell** [comment]
Yes, they can’t change it in Bundler 1.x, but `gems.rb` will likely be Bundler 2.0’s default: https://github.com/bundler/bundler/issues/694
